### ISSUE TITLE
Audio Loop Start and End Point: Allow Start Point to Come After End Points

### ISSUE THREAD

**Issue by ChrismuthMan** [YoYoGames/GameMaker-Bugs#8843]
### Is your feature request related to a problem?

Right now, the loop start has to be before the loop end, otherwise the time chosen gets overwritten. In order to skip to a later portion of a song (when the player reaches the second phase of a boss fight, for example), it would be useful to be able to set the loop end at the end of a measure, and the loop start at the start of the new section, which is further in the song. If loop end was set to 0:30, and loop start set to 1:00, then the song would skip directly to 1:00 when it reaches 0:30, then not loop again unless the loop start and end were changed. Currently, this is impossible, because it would be overwritten as it is not allowed by audio_sound_loop_start() and audio_sound_loop_end().

### Describe the solution you'd like

I simply request that setting the end point before the start point no longer force the end point to be placed at the very end of the sound. The following code:

audio_sound_loop_start(song,60);
audio_sound_loop_end(song,30);
audio_play_sound(song,100,true);

would play song starting at the beginning, then after 30 seconds, it would skip to the 60 second point if this change were implemented.
Right now, this code would set the loop end to 0, making the end of the song the loop end, so in a current project, this would play the entire song, then loop back to the 60 second point. In order to make sure this doesn't break old projects, the functions could have an extra variable:

audio_sound_loop_start(index,time,force)
audio_sound_loop_start(index,time,force)

This variable, force, would be set to false by default, and if false, it would work as it currently does, so that current projects aren't affected by the change. If set to true, then you can force the loop start to come before the loop end, allowing for the functionality that I have requested. 

If my other request (#8842) were implemented, then this variable and functionality would also be added to the new audio sync group functions:

audio_sync_group_loop_start(group_index,time,force)
audio_sync_group_loop_end(group_index,time,force)

Basically, instead of "The audio will loop between these two time stamps", it would be more like "When the audio reaches the end point, it will jump to the start point", regardless of the order of the two points.
